pbj-core/pbj-runtime/src/main/java/com/hedera/hashgraph/pbj/runtime/io/DataInput.java

```diff
@@ -0,0 +1,306 @@
+                    ((long) (b3 & 255) << 40) +
+                    ((long) (b4 & 255) << 32) +
+                    ((long) (b5 & 255) << 24) +
+                    ((b6 & 255) << 16) +
+                    ((b7 & 255) << 8) +
+                    ((b8 & 255)));
+        }
+    }
+
+    /**
+     * Reads the next four bytes at the current position, composing them into a float value according to the Java
+     * standard big-endian byte order, and then increments the position by four.
+     *
+     * @return The float value at the current position
+     * @throws BufferUnderflowException If there are fewer than four bytes remaining
+     * @throws IOException if an I/O error occurs
+     */
+    default float readFloat() throws IOException {
+        return Float.intBitsToFloat(readInt());
+    }
+
+    /**
+     * Reads the next four bytes at the current position, composing them into a float value according to specified byte
+     * order, and then increments the position by four.
+     *
+     * @param byteOrder the byte order, aka endian to use
+     * @return The float value at the current position
+     * @throws BufferUnderflowException If there are fewer than four bytes remaining
+     * @throws IOException if an I/O error occurs
+     */
+    default float readFloat(ByteOrder byteOrder) throws IOException {
+        return Float.intBitsToFloat(readInt(byteOrder));
+    }
+
+    /**
+     * Reads the next eight bytes at the current position, composing them into a double value according to the Java
+     * standard big-endian byte order, and then increments the position by eight.
+     *
+     * @return The double value at the current position
+     * @throws BufferUnderflowException If there are fewer than eight bytes remaining
+     * @throws IOException if an I/O error occurs
+     */
+    default double readDouble() throws IOException {
+        return Double.longBitsToDouble(readLong());
+    }
+
+    /**
+     * Reads the next eight bytes at the current position, composing them into a double value according to specified byte
+     * order, and then increments the position by eight.
+     *
+     * @param byteOrder the byte order, aka endian to use
+     * @return The double value at the current position
+     * @throws BufferUnderflowException If there are fewer than eight bytes remaining
+     * @throws IOException if an I/O error occurs
+     */
+    default double readDouble(ByteOrder byteOrder) throws IOException {
+        return Double.longBitsToDouble(readLong(byteOrder));
+    }
+
+    /**
+     * Read a 32bit protobuf varint at current position. An integer var int can be 1 to 5 bytes.
+     *
+     * @return integer read in var int format
+     * @param zigZag use protobuf zigZag varint encoding, optimized for negative numbers
+     * @throws IOException if an I/O error occurs
+     */
+    default int readVarInt(boolean zigZag) throws IOException {
+        return (int)readVarLong(zigZag);
```

Comment:
Is it safe to cast this? I think so... If you have two varInts in a row in the byte stream, is there any way it will be incorrectly interpreted as a single varlong with some data lost? I don't think it can be a problem, I think what you are doing here is safe.